### ISSUE TITLE
UI freezes while BG operations are performed

### ISSUE THREAD

**Issue by foxmask** [wallabag/android-app#315]
I made 2 videos to show you what's happened on our smarthone for Wallabag when the network does a "yo-yo" (go away / come back)

On the second one, you will see "black" or "white" screen.
This is not due to the video encoding. It's the real effect I could notice when the smartphone looses is breath. To fix this behavior I finally have to remove Wallabag from the recent app I launched.
You can see the time elasped in the footer of the video, showing you the real time I have to wait before being able to do anything

https://foxmask.trigger-happy.eu/static/wallabag/2016_10_04_07_40_18.mp4
https://foxmask.trigger-happy.eu/static/wallabag/2016_10_04_07_41_58.mp4

**Comment by Simounet** [comment]
I agree. With the last update, the app is not as responsive as it used to be. Each action seems to wait the wallabag response. It should be a background process with a notification about the response.

**Comment by di72nn** [comment]
The problem is long known, I even thought it was fixed in development (mentioned in #282 a couple of times).

I finally found the probable cause of this: DB transactions block any queries when the background service performs its operations (I keep the transaction in the background service active as long as the operation is performed). Sometimes UI manages to update itself before the service starts its operations, sometimes it gets blocked.

I'll try to fix it ASAP.
